nodejs/week3/warmup/calculator.js

```diff
@@ -0,0 +1,75 @@
+const e = require("express");
+const express = require("express");
+const router = express.Router();
+// const knex = require("../database");
+// const { request, response } = require("express");
+
+router.get("/", async(request, response) => {
+    try {
+        console.log("this is a calculator");
+        response.send("Let's get some calculations done");
+    } catch (error) {
+        throw error;
+    }
+});
+
+// GET section
+router.get("/add", (request, response) => {
+    const result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) + Number(b), 0);
+    response.send(`This is the result ${result}`);
+});
+
+router.get("/sub", (request, response) => {
+    const result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) - Number(b));
+    response.send(`This is the result ${result}`);
+});
+
+router.get("/multiply", (request, response) => {
+    let result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) * Number(b));
+    response.send(`This is the result ${result}`);
+});
+
+router.get("/divide", (request, response) => {
+    let result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) / Number(b));
+    response.send(`This is the result ${result}`);
+});
+
+// POST section
+
+router.post("/add", (request, response) => {
+    const result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) + Number(b), 0);
+    response.send(`This is the result ${result}`);
+});
+
+router.post("/sub", (request, response) => {
+    const result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) - Number(b));
+    response.send(`This is the result ${result}`);
+});
+
+router.post("/multiply", (request, response) => {
+    let result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) * Number(b));
+    response.send(`This is the result ${result}`);
+});
+
+router.post("/divide", (request, response) => {
+    let result = Object.values(request.query)
+        .flat()
+        .reduce((a, b) => Number(a) / Number(b));
+    response.send(`This is the result ${result}`);
+});
```

Comment:
These are post requests, but you're still accessing the query parameters, which are passed in as part of the url.
You need to read the parameters from the request body!
Here is some more information about the difference between GET and POST requests:
https://www.guru99.com/difference-get-post-http.html and https://www.postman.com/ is a useful tool to make POST requests and test your api.
Also, here is how you can access the body parameters in nodejs using express:
https://nodejs.dev/learn/get-http-request-body-data-using-nodejs
You basically need to do this:
```
app.use(
  express.urlencoded({
    extended: true
  })
)

app.use(express.json())

let result = Object.values(request.body)